### PR TITLE
Add table name prefixes in SELECT statment for fields

### PR DESCRIPTION
**Pull request by lafriks**
It was larger change than I expected initially but to me it looks like it's safest way to fix go-rel/rel#233

### CODE REVIEW

File: builder/buffer.go [hunk 1]
```diff
@@ -103,30 +106,43 @@ func (b *Buffer) WritePlaceholder() {
...
-	key := escapeCacheKey{value: value, quoter: b.Quoter}
+	key := escapeCacheKey{table: table, value: value, quoter: b.Quoter}
 	escapedValue, ok := escapeCache.Load(key)
 	if ok {
 		return escapedValue.(string)
 	}
 
 	if len(value) > 0 && value[0] == UnescapeCharacter {
 		escapedValue = value[1:]
+	} else if regexNumber.MatchString(value) {
```
Comment (lafriks) [inline builder/buffer.go:135]: This adds option to order/group by column number

Comment (Fs02) [reply builder/buffer.go:135]: will it improve performance if we change this to: `value[0] >= '0' && value[0] <= '9' && regexNumber.MatchString(value)`? 🤔

File: builder/buffer_test.go [hunk 2]
```diff
@@ -35,11 +37,26 @@ func TestBuffer_escape(t *testing.T) {
 			field:  "user.*",
 			result: "[user].*",
 		},
+		{
```
Comment (Fs02) [inline builder/buffer_test.go:40]: how about test case when table name already specified in value? (for example when comparing join field)

```
{
  table: "user",
  field: "address.name",
  result: "[address].[name]"
}
```

File: builder/delete.go [hunk 3]
```diff
@@ -22,7 +22,7 @@ func (ds Delete) Build(table string, filter rel.FilterQuery) (string, []interfac
 
 	if !filter.None() {
 		buffer.WriteString(" WHERE ")
-		ds.Filter.Write(&buffer, filter, ds.Query)
+		ds.Filter.Write(&buffer, "", filter, ds.Query)
```
Comment (Fs02) [inline builder/delete.go:25]: why not specify table as well here?

File: builder/update.go [hunk 4]
```diff
@@ -51,7 +51,7 @@ func (u Update) Build(table string, primaryField string, mutates map[string]rel.
 
 	if !filter.None() {
 		buffer.WriteString(" WHERE ")
-		u.Filter.Write(&buffer, filter, u.Query)
+		u.Filter.Write(&buffer, "", filter, u.Query)
```
Comment (Fs02) [inline builder/update.go:54]: same question with delete

**Pull request merged**